processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -285,31 +311,53 @@ void add(@Nullable final Filter filter)
             }
           }
         }
+        preFilters.add(filter);
+
+      }
+    }
 
-        if (requiredColumns.contains(outputColumnName)) {
-          // Rewrite filter post-unnest if possible.
+    final FilterSplitter filterSplitter = new FilterSplitter();
+
+    // non-unnest case
+    // OR CASE
+    List<Filter> preFilterList = new ArrayList<>();
+    List<Filter> postFilterList = new ArrayList<>();
```

Comment:
The `preFilterList`, `postFilterList` here is very confusing and easy to mix up with the `filterSplitter`...  I'm not sure how to separate it exactly, but I almost thought that these were getting AND'd together because I confused them with the FilterSplitter filters.